### PR TITLE
Remove the process agent kitchen tests

### PR DESCRIPTION
**Pull request by FlorentClarret**
### What does this PR do?

Remove the process agent kitchen tests

### Motivation

Follow-ups for https://app.datadoghq.com/incidents/26770, these tests are not needed anymore as superseded by e2e tests. 

We only dropped the one that was failing in a previous PR: https://github.com/DataDog/datadog-agent/pull/24778

### CODE REVIEW

**Review by chouquette — APPROVED** [review]
LGTM assuming the pipeline passes :)

**Review by chouetz — APPROVED** [review]
As long as the remaining stuff works I'm fine with this change!

File: .gitlab/maintenance_jobs/kitchen.yml
Comment (KSerrania) [inline .gitlab/maintenance_jobs/kitchen.yml:42]: I'd do so in a separate step, a few days later, otherwise the resources created just before the merge of this PR may never get cleaned up.

Comment (FlorentClarret) [reply .gitlab/maintenance_jobs/kitchen.yml:42]: Oh I was not aware of this. Let me revert and I'll open another PR in a couple of days. Thank you

**Review by pducolin — APPROVED** [review]

**Review by robertjli — APPROVED** [review]

**Pull request merged**
